lib/abstract_format.ml

```diff
@@ -596,6 +603,56 @@ and expr_of_sf sf : (expr_t, err_t) Result.t =
      let%bind rhs = sf_rhs |> expr_of_sf |> track ~loc:[%here] in
      ExprBinOp {line; op; lhs; rhs} |> return
 
+  (* a record creation : #user{name = "Taro", admin = true} *)
+  | Sf.Tuple (4, [Sf.Atom "record";
+                  Sf.Integer line;
+                  Sf.Atom name;
+                  Sf.List sf_record_fields]) ->
+     let field_of_sf sf =
+       begin match record_field_of_sf sf with
+       | Ok (RecordField {line; field_name; ty=None; default_expr=Some e}) ->
+          (line, field_name, e) |> return
+       | Ok _ -> failwith "cannot reach here"
```

Comment:
Could you use [`Err.Invalid_input`](https://github.com/yutopp/obeam/blob/d4974b8f2542d88cf14e89f33dc2bdba1bcfcdec/lib/err.ml#L18)?

Reply:
fixed.